File: backend/app/application/dto/baby_dto.py
from dataclasses import dataclass
from datetime import date, datetime
from uuid import UUID


@dataclass
class CreateBabyDTO:
    user_id: UUID
    name: str
    birth_date: date
    gender: str | None = None
    birth_weight_g: int | None = None
    birth_height_cm: float | None = None
    birth_head_circumference_cm: float | None = None
    birth_chest_circumference_cm: float | None = None
    blood_type: str | None = None
    rh_factor: str | None = None
    birth_time: str | None = None


@dataclass
class UpdateBabyDTO:
    id: UUID
    name: str | None = None
    birth_date: date | None = None
    gender: str | None = None
    birth_weight_g: int | None = None
    photo_url: str | None = None
    birth_height_cm: float | None = None
    birth_head_circumference_cm: float | None = None
    birth_chest_circumference_cm: float | None = None
    blood_type: str | None = None
    rh_factor: str | None = None
    birth_time: str | None = None


@dataclass
class BabyResponseDTO:
    id: UUID
    user_id: UUID
    name: str
    birth_date: date
    gender: str | None
    birth_weight_g: int | None
    age_days: int
    age_months: int
    created_at: datetime
    photo_url: str | None = None
    birth_height_cm: float | None = None
    birth_head_circumference_cm: float | None = None
    birth_chest_circumference_cm: float | None = None
    blood_type: str | None = None
    rh_factor: str | None = None
    birth_time: str | None = None

File: zzippu_ios/zzippu/Data/Network/DTOs/BabyDTO.swift
// Data/Network/DTOs/BabyDTO.swift
// 서버 BabyResponse / BabyCreateRequest / BabyUpdateRequest 대응

import Foundation

// MARK: - Response DTO

struct BabyResponseDTO: Decodable {
    let id: UUID
    let userId: UUID?
    let name: String
    let birthDate: String          // date(YYYY-MM-DD) — Mapper에서 변환
    let birthTime: String?         // "HH:mm"(24h, KST) — Mapper에서 birthDate와 합침. nullable.
    let gender: String?
    let birthWeightG: Int?
    let birthHeightCm: Double?     // 출생 측정치(선택)
    let birthHeadCircumferenceCm: Double?
    let birthChestCircumferenceCm: Double?
    let bloodType: String?         // A/B/O/AB (선택)
    let rhFactor: String?          // positive/negative (선택)
    let ageDays: Int?              // 서버 파생값 (iOS에서 무시)
    let ageMonths: Int?            // 서버 파생값 (iOS에서 무시)
    let photoUrl: String?
    let createdAt: Date            // datetime — 자동 디코딩
}

// MARK: - Create Request DTO

struct BabyCreateRequestDTO: Encodable {
    let name: String
    let birthDate: String          // YYYY-MM-DD
    let birthTime: String?         // "HH:mm"(24h, KST) — 선택
    let gender: String?
    let birthWeightG: Int?
    let birthHeightCm: Double?
    let birthHeadCircumferenceCm: Double?
    let birthChestCircumferenceCm: Double?
    let bloodType: String?         // A/B/O/AB
    let rhFactor: String?          // positive/negative
}

// MARK: - Update Request DTO

struct BabyUpdateRequestDTO: Encodable {
    let name: String?
    let birthDate: String?         // YYYY-MM-DD
    let birthTime: String?         // "HH:mm"(24h, KST) — 선택
    let gender: String?
    let birthWeightG: Int?
    let birthHeightCm: Double?
    let birthHeadCircumferenceCm: Double?
    let birthChestCircumferenceCm: Double?
    let bloodType: String?         // A/B/O/AB
    let rhFactor: String?          // positive/negative
    let photoUrl: String?
}

// MARK: - Caregiver Join Request DTO

struct CaregiverJoinRequestDTO: Encodable {
    let code: String
}
